OpenRA.Mods.Common/Traits/World/RuntimeRulesetReload.cs

```diff
@@ -0,0 +1,117 @@
+	public class RuntimeRulesetReloadInfo : TraitInfo
+	{
+		public override object Create(ActorInitializer init)
+		{
+			return new RuntimeRulesetReload();
+		}
+	}
+
+	public class RuntimeRulesetReload : IChatCommand, IWorldLoaded, IPostWorldLoaded, ITick
+	{
+		[TranslationReference]
+		const string CheatsDisabled = "notification-cheats-disabled";
+
+		[TranslationReference]
+		const string CommandDescription = "description-ruleset-autoreload";
+
+		const string CommandName = "autoreload";
+
+		[TranslationReference]
+		const string AutoReloadEnabled = "notification-ruleset-autoreload-enabled";
+
+		[TranslationReference]
+		const string AutoReloadDisabled = "notification-ruleset-autoreload-disabled";
+
+		[TranslationReference]
+		const string CommandArgumentError = "notification-ruleset-autoreload-invalid-argument";
+
+		[TranslationReference]
+		const string AutoReloadMultiplayerDisabled = "notification-ruleset-autoreload-multiplayer-disabled";
+
+		World world;
+		DeveloperMode developerMode;
+
+		public bool Enabled
+		{
+			get => Game.Settings.Debug.EnableRulesetAutoReload;
+			set => Game.Settings.Debug.EnableRulesetAutoReload = value;
+		}
+
+		void IWorldLoaded.WorldLoaded(World w, WorldRenderer wr)
+		{
+			world = w;
+
+			if (world.LocalPlayer != null)
+				developerMode = world.LocalPlayer.PlayerActor.Trait<DeveloperMode>();
+
+			var console = w.WorldActor.TraitOrDefault<ChatCommands>();
+			var help = w.WorldActor.TraitOrDefault<HelpCommand>();
+
+			if (console == null || help == null)
+				return;
+
+			console.RegisterCommand(CommandName, this);
+			help.RegisterHelp(CommandName, CommandDescription);
+		}
+
+		void IPostWorldLoaded.PostWorldLoaded(World w, WorldRenderer wr)
+		{
+			if (Enabled && w.GameInfo.IsSinglePlayer)
+				Game.RulesetWatcher.StartWatching();
+		}
+
+		void IChatCommand.InvokeCommand(string name, string arg)
+		{
+			if (world.LocalPlayer == null)
+				return;
+
+			if (!developerMode.Enabled)
+			{
+				TextNotificationsManager.Debug(TranslationProvider.GetString(CheatsDisabled));
+				return;
+			}
+
+			if (!world.GameInfo.IsSinglePlayer)
+			{
+				TextNotificationsManager.Debug(TranslationProvider.GetString(AutoReloadMultiplayerDisabled));
+				return;
+			}
+
+			if (name == CommandName)
+			{
+				bool shouldEnable;
+				if (string.IsNullOrEmpty(arg))
+					shouldEnable = !Enabled;
+				else if (!bool.TryParse(arg, out shouldEnable))
+				{
+					TextNotificationsManager.Debug(TranslationProvider.GetString(CommandArgumentError));
+					return;
+				}
+
+				Enabled = shouldEnable;
+
+				TextNotificationsManager.Debug(TranslationProvider.GetString(shouldEnable ? AutoReloadEnabled : AutoReloadDisabled));
+			}
+		}
+
+		void ITick.Tick(Actor self)
+		{
+			var isEnabled = Enabled;
+			if (Game.RulesetWatcher.IsEnabled == isEnabled)
+				return;
+
+			if (isEnabled)
+				Game.RulesetWatcher.StartWatching();
+			else
+				Game.RulesetWatcher.StopWatching();
```

Comment:
As mentioned in our chat, I think we can get rid of this while still keeping `RulesetWatcher` independent. If you can have a `ToggleWatching(bool toggleValue)` method that can do both starting and stopping (as these are similar enough to be one method), and call it when the checkbox is ticked _or_ the `/autoreload` command is used that would make this simpler plus easier to understand.